src/propagate-tor-test_cuda.cu

```diff
@@ -918,39 +924,73 @@ int main (int argc, char* argv[]) {
 
    printf("done preparing!\n");
  
-   printf("Number of struct MPTRK trk[] = %ld\n", nevts*nb);
-   printf("Number of struct MPTRK outtrk[] = %ld\n", nevts*nb);
-   printf("Number of struct struct MPHIT hit[] = %ld\n", nevts*nb);
+   printf("Number of struct MPTRK trk[] = %d\n", nevts*nb);
+   printf("Number of struct MPTRK outtrk[] = %d\n", nevts*nb);
+   printf("Number of struct struct MPHIT hit[] = %d\n", nevts*nb);
   
    printf("Size of struct MPTRK trk[] = %ld\n", nevts*nb*sizeof(struct MPTRK));
    printf("Size of struct MPTRK outtrk[] = %ld\n", nevts*nb*sizeof(struct MPTRK));
    printf("Size of struct struct MPHIT hit[] = %ld\n", nlayer*nevts*nb*sizeof(struct MPHIT));
 
-   
+   auto doWork = [&](const char* msg, int nIters) {
+     double wall_time = 0;
+
+#ifdef MEASURE_H2D_TRANSFER
+     for(int itr=0; itr<nIters; itr++) {
+       auto wall_start = std::chrono::high_resolution_clock::now();
+       for (int s = 0; s<num_streams;s++) {
+         transferAsyncTrk(trk_dev, trk,streams[s]);
+         transferAsyncHit(hit_dev, hit,streams[s]);
+       }
+
+       for (int s = 0; s<num_streams;s++) {
+         printf("%s ... <<<%i,%i>>>\n", msg, (nevts*ntrks)/threadsperblock+1,threadsperblock);
+         GPUsequence<<<nevts*nb/threadsperblock+1,threadsperblock ,0,streams[s]>>>(trk_dev,hit_dev,outtrk_dev,s);
+       }
+
+#ifdef MEASURE_D2H_TRANSFER
+       for (int s = 0; s<num_streams;s++) {
+         transfer_backAsync(outtrk, outtrk_dev,streams[s]);
```

Comment:
This will imply we only get the output tracks if we do `MEASURE_D2H_TRANSFER`, right?
Is this the behavior that we want?

Reply:
Good point, I'll add that back in the end.